stdlib/3/smtplib.pyi

```diff
@@ -47,24 +48,32 @@ class SMTPAuthenticationError(SMTPResponseException): ...
 def quoteaddr(addrstring: str) -> str: ...
 def quotedata(data: str) -> str: ...
 
+class _AuthObject(Protocol):
+    @overload
+    def __call__(self, challenge: None = ...) -> Optional[str]: ...
+    @overload
+    def __call__(self, challenge: bytes) -> str: ...
```

Comment:
`authobject()` is documented to return `bytes`, but this neither matches the documentation nor the return values of `auth_cram_md5()` et al.

Reply:
Filed as [bpo-34525](https://bugs.python.org/issue34525) and PR python/cpython#8965.